### PR TITLE
update asto library to 0.16.1

### PR DESCRIPTION
**Pull request by brastak**
For #52 and #53 - we need to use asto 0.16+ where deletion was implemented
Pay attention - Rpm API was changed:

- update(String) -> update(Key)
- batchUpdate(String) -> batchUpdate(Key)

It was necessary because of batchUpdate must works with Key.ROOT that could not be created from String.

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#54](https://github.com/artipie/rpm-adapter/issues/54) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/ebc3f480-8d16-4690-8bd0-2c713883941c); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by g4s8** [comment]
@0crat assign me

**Comment by 0crat** [comment]
> @0crat assign me ([here](https://github.com/artipie/rpm-adapter/issues/54#issuecomment-604282460))

@g4s8 This pull request [#54](https://github.com/artipie/rpm-adapter/issues/54) is assigned to @g4s8[/z](https://www.0crat.com/u/g4s8), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/8440982d-3e5d-4dde-810b-be26f733ca9c); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/main/java/com/artipie/rpm/Rpm.java [hunk 1]
```diff
@@ -89,19 +95,23 @@
...
     /**
      * Ctor.
      * @param stg The storage
+     * @param vertx The Vertx instance
      * @param naming RPM files naming policy
      * @param dgst Hashing sum computation algorithm
+     * @checkstyle ParameterNumberCheck (10 lines)
      */
-    public Rpm(final Storage stg, final NamingPolicy naming, final Digest dgst) {
+    public Rpm(final Storage stg, final Vertx vertx, final NamingPolicy naming, final Digest dgst) {
```
Comment (g4s8) [inline src/main/java/com/artipie/rpm/Rpm.java]: @brastak same here: please keep `final Storage stg, final NamingPolicy naming, final Digest dgst` signature too

File: src/main/java/com/artipie/rpm/Rpm.java [hunk 2]
```diff
@@ -115,18 +125,23 @@ public Rpm(final Storage stg, final NamingPolicy naming, final Digest dgst) {
      * @param key The name of the file just updated
      * @return Completion or error signal.
      */
-    public Completable update(final String key) {
+    public Completable update(final Key key) {
```
Comment (g4s8) [inline src/main/java/com/artipie/rpm/Rpm.java]: @brastak same here: you can wrap this method with:
```java
public Completable update(final String key) {
  return this.update(new Key.From(key));
}
```

File: src/main/java/com/artipie/rpm/Rpm.java [hunk 3]
```diff
@@ -156,14 +171,13 @@ public Completable update(final String key) {
 
     /**
      * Batch update RPM files for repository.
-     * @param repo Repository key
+     * @param prefix Repository key prefix
      * @return Completable action
      */
-    public Completable batchUpdate(final String repo) {
-        return SingleInterop.fromFuture(this.storage.list(new Key.From(repo)))
+    public Completable batchUpdate(final Key prefix) {
```
Comment (g4s8) [inline src/main/java/com/artipie/rpm/Rpm.java]: @brastak same here (method signature)

**Comment by brastak** [comment]
@g4s8 Please, see update

**Comment by 0crat** [comment]
Job was finished in 2 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Pull request merged**